teps/0053-nested-triggers.md

```diff
@@ -201,4 +210,121 @@ spec:
+
+### Option 2: Interceptor Groups
+
+Specify the interceptor groups at the top level and process each group on trigger:
+
+```yaml
+apiVersion: triggers.tekton.dev/v1alpha1
+kind: EventListener
+metadata:
+  name: my-el
+spec:
+  groups: 
+  - name: github-group
+    interceptors:
+      - my-interceptor
+      - add-extra-body
+    triggerSelector:
+      matchLabels:
+        app=foo
+        anotherval=xyz
+  - name: another-group
+    interceptors:
+      - my-interceptor
+    triggerSelector:
+       matchLabels:
+         app=foo
+```
+
+This can work well with running a root EventListener interceptor with a `labelSelector`. One challenge here is that these `InterceptorGroups` become another separate API resource that would need to be managed and potentially added to the API. This could be added onto the existing Trigger resource, but then it becomes closer to Option 1.
+
+#### Advantages
+
+* EventListener Readability - the list of interceptor groups sits in the EventListener objects, so when adding a trigger, a trigger author can check in there, and decide which labels to add to it so that it ends up under the correct top trigger.
+* Resource Trigger Validation - Provides the same advantage from Option 1.
+* Implementation within the current system - this implementation could be done with minimal impact to the API by wrapping the trigger invocation by a group processing endpoint.
+
+#### Disadvantages
+
+* Trigger Complexity - since this only allows a single fan out, the downstream trigger for each one of these trigger resources would need to validate/assemble all data required to determine what resources should be generated by the eventlistener.
+* Nesting Limitations - This creates a graph with a single edge, from the groups to the target triggers. This limits the ability to chain resources together to complete multiple "common validation" steps before determining resource targets.
+
+
+### Option 3: Downstream EventListener
+
+Allow triggers to target other eventListeners after processing interceptors:
+
+```yaml
+apiVersion: triggers.tekton.dev/v1alpha1
+kind: Trigger
+metadata:
+  name: top-level-trigger
+spec:
+  interceptors:
+    - cel:
+        filter: >-
+          header.match('header-filter', 'check-this-header-filter')
+        - key: example-overlay
+          expression: body.example.value
+    - webhook:
+        objectRef:
+          kind: Service
+          name: example-webhook-service
+          apiVersion: v1
+          namespace: tekton-pipelines
+  eventListeners:
+  #Two possible ways of specifying target
+  - target-namespace/downstream-eventlistener
+  - http://downstream-eventlistener.target-namespace.svc:8080
+```
+
```

Comment:
Another alternative: could we use an admission webhook to enforce that triggers always start with a certain set of interceptors?

Reply:
This is a challenge because that config lives at the cluster level. If the cluster has multiple ELs, then we are enforcing a scoped behavior (scoped to a trigger object referenced by one or more ELs) at the cluster level.